docs: `docs/reference` shortcode for dummies (#271)
grafana/writers-toolkit: docs/sources/writing-guide/shortcodes/index.md

```

**Reference:**
```markdown
{{%/* docs/reference */%}}
[1]: "http://google.com/" 
[2]: "http://search.yahoo.com/" 
[3]: "http://search.msn.com/"
{{%/* /docs/reference */%}}
```

This shortcode is useful when:

- You want to link to the same destination multiple times, because it saves you time formatting the link every time.
- When content from one repository is published to more than one documentation set, because the `docs/reference` shortcode lets you specify appropriate links for each doc set.

For example, alerting documentation is sourced from the Grafana repository and published to both Grafana documentation and Grafana Cloud documentation.
Any link from alerting documentation to dashboard documentation, for example, which is also published to both Grafana and Grafana Cloud, needs to be specific as to which destination is appropriate for the product documentation.
Grafana Cloud alerting documentation should link to Grafana Cloud dashboard documentation, and Grafana alerting documentation should link to Grafana dashboard documentation.

The following content must be within the opening and closing tags of the `docs/reference` shortcode:

>  `label` - The label for the shortcode. This is the text that you enter in the link instead of an http link or file path.  

Reviewer: Do we need to be careful to say that the label uses `[<LABEL>]` syntax whereas a "usual" link uses `(<DESTINATION>)` syntax.

Most importantly that the label is in square brackets and the destination uses parentheses.

nit: I'm not sure if we should be trying to be consistent with the command line placeholder style guidance: https://grafana.com/docs/writers-toolkit/write/style-guide/write-for-developers/#command-lines

For example, a page in versioned Grafana documentation is also mounted in the Grafana Cloud documentation. 
The page in Grafana should link to the Grafana dashboards page and the page in Grafana Cloud should link to the Grafana Cloud dashboards page.

Set the reference at the end of the document as follows:

```markdown
{{%/* docs/reference */%}}
[dashboards]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/dashboards"
[dashboards]: "/docs/grafana-cloud/ -> /docs/grafana-cloud/dashboards"
{{%/* /docs/reference */%}}
```

The content within the shortcode tags is as follows:

 `label` - The label for the shortcode. This is the text that you'll use in the body text. In the example above, the label is `dashboards`. The label can be multiple words (for example, [dashboards docs]).
 `project path prefix` - Designates the destination project. In the example above, the path prefixes are `/docs/grafana/` for Grafana and `/docs/grafana-cloud/` for Cloud.  

>  `reference` - The path to the file. It can include `<SOMETHING VERSION>`, which is either taken from front matter of (which file) or falls back to being inferred from the version of the page. This enables the use of absolute paths that resolve correctly, irrespective of version.

Author: ```suggestion
 `reference` - The path to the file. It can include `<SOMETHING VERSION>`, which is either taken from front matter of (which file) or falls back to being inferred from the version of the page. This enables the use of absolute paths that resolve correctly, irrespective of version. When including a version, for the target project, use the name of the project, with spaces but no hyphens or underscores, all uppercased (for example, grafana = GRAFANA, grafana-cloud = GRAFANA CLOUD).
```

<!--TO DO
Question: what would be the "something" be in a different repo? is it the name of the repo?-->

Then add the link in the body of the file in the following format:

```markdown
For more information about Grafana dashboards, refer to the [Dashboards documentation][dashboards].
```

- If the page you're on is `/docs/grafana/latest/alerting`, the inferred version is `latest`, and the returned reference is `/docs/grafana/latest/dashboards`.

> - If the page you're on is `/docs/grafana/next/alerting`, the inferred version is `next`, and the returned reference is `/docs/grafana/next/dashboards`.

Reviewer: nit: Hugo pages have a trailing slash. Hugo relrefs do not though.

```suggestion
- If the page you're on is `/docs/grafana/next/alerting/`, the inferred version is `next`, and the returned reference is `/docs/grafana/next/dashboards`.
```

You can override version inference using front matter.
To override the value of `<GRAFANA VERSION>`, set the `grafana_version` parameter in the page's front matter.
For example, with the front matter `grafana_version: next`, the shortcode replaces `<GRAFANA VERSION>` with `next`.

### Other use cases

The `docs/reference` shortcode is also useful when you want to link to the same destination multiple times in one file. 
It allows you to specify the link destination once while you use the label multiple times. For example:

**Reference:**

```markdown
{{%/* docs/reference */%}}

> [Grafana website]: "www.grafana.com"

Reviewer: ```suggestion
[Grafana website]: "/ -> www.grafana.com"
```

 `project path prefix` - Designates the destination project. For example, for Grafana, use `/docs/grafana/`, for Mimir, use `/docs/mimir/`.   
 `reference` - The path to the file. It can include `<SOMETHING VERSION>`, which is either taken from front matter of (which file) or falls back to being inferred from the version of the page. This enables the use of absolute paths that resolve correctly, irrespective of version.                                                                                                                                                          

Set the docs reference in the footer of the page and then use the `label` in the link in the body of the file.

### Example

In the following example, a page in versioned Grafana documentation is also mounted in the Grafana Cloud documentation.

The page in Grafana should link to the Grafana dashboards page and the page in Grafana Cloud should link to the Grafana Cloud dashboards page.

```markdown
{{%/* docs/reference */%}}
[dashboards]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/dashboards"
[dashboards]: "/docs/grafana-cloud/ -> /docs/grafana-cloud/dashboards"
{{%/* /docs/reference */%}}
```

> 

Author: ```suggestion
```

| `label`               | The label for the shortcode. This is the text that you enter in the link instead of an http link or file path.                                                                                                                                                            | yes      |
| `project path prefix` | Designates the destination project. For example, for Grafana, use `/docs/grafana/`, for Mimir, use `/docs/mimir/`.                                                                                                                                                        | yes      |
| `reference`           | The path to the file. It can include `<SOMETHING VERSION>`, which is either taken from front matter of (which file) or falls back to being inferred from the version of the page. This enables the use of absolute paths that resolve correctly, irrespective of version. | yes      |

Set the docs reference in the footer of the page and then use the `label` in the link in the body of the file.

> ### Examples

Author: ```suggestion
### Example
```

{{</* section withDescriptions="true"*/>}}
```

## `docs/reference` shortcode

The `docs/reference` shortcode offers more flexible linking than the Hugo builtin `relref` shortcode.
When content from one repository is published to more than one documentation set, the `docs/reference` shortcode lets you specify appropriate links for each doc set.

For example, alerting documentation is sourced from the Grafana repository and published to both Grafana documentation and Grafana Cloud documentation.
Any link from alerting documentation to dashboard documentation, for example, which is also published to both Grafana and Grafana Cloud, needs to be specific as to which destination is appropriate for the product documentation.
Grafana Cloud alerting documentation should link to Grafana Cloud dashboard documentation, and Grafana alerting documentation should link to Grafana dashboard documentation.

The content of the `docs/reference` shortcode must be within opening and closing tags.

> | Parameter             | Description                                                                                                                                                                                                                                                               | Required |

Reviewer: I don't think this table is parallel with the others in the page.

All others are talking specifically about Hugo shortcode parameters. This table is talking about elements of the text within the opening and closing shortcode tags.

Reviewer: I think we need a usage example with placeholder text so that readers can understand what `label` or the other rows are referring to.

> | Parameter             | Description                                                                                                                                                                                                                                                               | Required |

Author: ```suggestion
```

| --------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------- |

> | `label`               | The label for the shortcode. This is the text that you enter in the link instead of an http link or file path.                                                                                                                                                            | yes      |

Reviewer: I don't think this will makes sense to somebody who hasn't yet seen a reference style link. There's also nothing to help a reader link this to the text in the example. That is, nothing in the example is obviously a label unless you know about reference style links.

Additionally, I think "instead of an HTTP link or file path" might be confusing. I think that without understanding reference style links, I would replace `[text](http://example.com)` incorrectly with `[text](label)` instead of `[text][label]`.

> | --------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------- |
> | `label`               | The label for the shortcode. This is the text that you enter in the link instead of an http link or file path.                                                                                                                                                            | yes      |

Author: ```suggestion
 `label` - The label for the shortcode. This is the text that you enter in the link instead of an http link or file path.  
 `project path prefix` - Designates the destination project. For example, for Grafana, use `/docs/grafana/`, for Mimir, use `/docs/mimir/`.   
 `reference` - The path to the file. It can include `<SOMETHING VERSION>`, which is either taken from front matter of (which file) or falls back to being inferred from the version of the page. This enables the use of absolute paths that resolve correctly, irrespective of version.                                                                                                                                                          
```

| `project path prefix` | Designates the destination project. For example, for Grafana, use `/docs/grafana/`, for Mimir, use `/docs/mimir/`.                                                                                                                                                        | yes      |

> | `reference`           | The path to the file. It can include `<SOMETHING VERSION>`, which is either taken from front matter of (which file) or falls back to being inferred from the version of the page. This enables the use of absolute paths that resolve correctly, irrespective of version. | yes      |

Reviewer: I still think it is important to reference the relref parameters as described in Hugo documentation. To me, the reference in the example below does not look like "the path to the file" because it has no file extension. It looks more like a path to a page, but that is also not correct and will cause confusion if people adopt that mental model. It is strictly the same as the parameter described in https://gohugo.io/content-management/cross-references/#use-of-ref-and-relref.

> | `project path prefix` | Designates the destination project. For example, for Grafana, use `/docs/grafana/`, for Mimir, use `/docs/mimir/`.                                                                                                                                                        | yes      |
> | `reference`           | The path to the file. It can include `<SOMETHING VERSION>`, which is either taken from front matter of (which file) or falls back to being inferred from the version of the page. This enables the use of absolute paths that resolve correctly, irrespective of version. | yes      |

Author: ```suggestion
```

{{</* section withDescriptions="true"*/>}}
```

## `docs/reference` shortcode

The `docs/reference` shortcode offers more flexible linking than the Hugo built-in `relref` shortcode.

Use this code when content from one repository is published to more than one documentation set, because it lets you specify appropriate links for each doc set in one part of the file (usually at end of the file, like a footer) while using the link label in the body text.

For example, a page in versioned Grafana documentation is also mounted in the Grafana Cloud documentation.
The page in Grafana should link to the Grafana dashboards page and the page in Grafana Cloud should link to the Grafana Cloud dashboards page.

Set the reference at the end of the document as follows:

```markdown
{{%/* docs/reference */%}}
[dashboards]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/dashboards"
[dashboards]: "/docs/grafana-cloud/ -> /docs/grafana-cloud/dashboards"
{{%/* /docs/reference */%}}
```

The content within the shortcode tags is as follows:

> - `label` - The label for the shortcode. This is the text that you'll use in the body text. In the example above, the label is `dashboards`. The label can be multiple words (for example, [dashboard docs]) and can include spaces.

Author: @jdbaldry - I'm good with this suggestion. Are you?

> - `project path prefix` - Designates the destination project. In the example above, the path prefixes are `/docs/grafana/` for Grafana and `/docs/grafana-cloud/` for Cloud.  

Reviewer 2: Are these paths in the `website` repo?  If yes, it would be helpful to document that.

Author: The Cloud project is in the `website` repo, but the Grafana one isn't.
